spec/slack-ruby-bot-server/app_spec.rb

```diff
@@ -13,6 +13,41 @@
       expect(app.instance).to be_an_instance_of(app)
     end
   end
+  context '#prepare!' do
+    context 'when connection cannot be established' do
+      context 'with ActiveRecord >= 7.2' do
```

Comment:
added a spec for 7.2+ vs pre-7.2.

quick explanation: 

`DatabaseCleaner` leases the connection, so `ActiveRecord::Base.connected?` is `true` when the spec starts, but we want to simulate what should happen in prod:
1. `ActiveRecord::Base.connected?` is false
2. `ActiveRecord::Base.connection.database_exists?` (or `ActiveRecord::Base.connection_pool.with_connection(&:active?)`) connects to the database
3. `ActiveRecord::Base.connected?` is true

in pre-7.2 that does not seem to work, the specs ultimately fail because `DatabaseCleaner` tries to operate on a closed connection. i've not been able to come up with a good spec for pre-7.2 so not sure if we should keep the spec, lmkwyt!

i added comments as well because it's maybe not obvious what is going on

---

along the way i had to do the following changes
1. pin `pagy_cursor` to `~> 0.6.1`
    + the method `pagy_cursor` only supports 2 parameters on 0.8, see [here](https://github.com/Uysim/pagy-cursor/blob/104c9cb2a15ac5e9cdd75e8ac276173a998d5a7a/lib/pagy_cursor/pagy/extras/cursor.rb#L6)
1. pin `database_cleaner` to a higher version to support ActiveRecord 7.2
    + ran into https://github.com/DatabaseCleaner/database_cleaner-active_record/issues/83

i ran the full spec suite locally with all versions from the matrix and all passed, so let's hope ci does us the honor as well

Reply:
> i ran the full spec suite locally with all versions from the matrix and all passed, so let's hope ci does us the honor as well

First time contributor PRs marked as pending is a little annoying (a 1-time problem), but you can always enable CI on your fork for next time you run into this.

Reply:
> but you can always enable CI on your fork

thanks for the tip, i somehow overlooked this lol